SSA/Projects/RISCV64/PrettyEDSL.lean

```diff
@@ -0,0 +1,185 @@
+import SSA.Core.MLIRSyntax.PrettyEDSL
+import SSA.Projects.RISCV64.Base
+import SSA.Projects.RISCV64.Syntax
+import SSA.Projects.InstCombine.Tactic
+import SSA.Projects.InstCombine.TacticAuto
+import SSA.Projects.InstCombine.Base
+import SSA.Projects.InstCombine.ForLean
+/-!
+This file defines a pretty printing/syntax for the `RISCV64` dialect.
+This allows the dialect to be written in MLIR-style syntax as well as
+assembly-like style.
+
+[RV64_com| {
+  ^bb0(%r1 : !i64, %r2 : !i64 ):
+  %1 =  const (0) : !i64
+  %2 =  sub %r1,  %1 : !i64
+        ret %2 : !i64
+
+We use the functionalities defined in `SSA.Core.MLIRSyntax.PrettyEDSL`.
+For operands with immediates or shift amounts encoded as part of the opcode,
+we define additional rules.
+ -/
+
+open MLIR.AST
+open Lean
+
+syntax "add" : MLIR.Pretty.uniform_op
+syntax "sub" : MLIR.Pretty.uniform_op
+syntax "and" : MLIR.Pretty.uniform_op
+syntax "or" : MLIR.Pretty.uniform_op
+syntax "xor" : MLIR.Pretty.uniform_op
+syntax "sll" : MLIR.Pretty.uniform_op
+syntax "sra" : MLIR.Pretty.uniform_op
+syntax "mul" : MLIR.Pretty.uniform_op
+syntax "div" : MLIR.Pretty.uniform_op
+syntax "divu" : MLIR.Pretty.uniform_op
+syntax "remu" : MLIR.Pretty.uniform_op
+syntax "rem" : MLIR.Pretty.uniform_op
+syntax "li" : MLIR.Pretty.uniform_op
+syntax "ret" : MLIR.Pretty.uniform_op
+syntax "sext.b" : MLIR.Pretty.uniform_op
+syntax "sext.h" : MLIR.Pretty.uniform_op
+syntax "zext.h" : MLIR.Pretty.uniform_op
+syntax "ror" : MLIR.Pretty.uniform_op
+syntax "rol" : MLIR.Pretty.uniform_op
+syntax "addw" : MLIR.Pretty.uniform_op
+syntax "subw" : MLIR.Pretty.uniform_op
+syntax "sllw" : MLIR.Pretty.uniform_op
+syntax "srlw" : MLIR.Pretty.uniform_op
+syntax "sraw" : MLIR.Pretty.uniform_op
+syntax "srl" : MLIR.Pretty.uniform_op
+syntax "slt" : MLIR.Pretty.uniform_op
+syntax "sltu" : MLIR.Pretty.uniform_op
+syntax "czero.eqz" : MLIR.Pretty.uniform_op
+syntax "czero.nez" : MLIR.Pretty.uniform_op
+syntax "bclr" : MLIR.Pretty.uniform_op
+syntax "bext" : MLIR.Pretty.uniform_op
+syntax "binv" : MLIR.Pretty.uniform_op
+syntax "bset" : MLIR.Pretty.uniform_op
+syntax "rolw" : MLIR.Pretty.uniform_op
+syntax "rorw" : MLIR.Pretty.uniform_op
+syntax "mulu" : MLIR.Pretty.uniform_op
+syntax "mulh" : MLIR.Pretty.uniform_op
+syntax "mulhu" : MLIR.Pretty.uniform_op
+syntax "mulhsu" : MLIR.Pretty.uniform_op
+syntax "mulw" : MLIR.Pretty.uniform_op
+syntax "divw" : MLIR.Pretty.uniform_op
+syntax "divwu" : MLIR.Pretty.uniform_op
+syntax "remw" : MLIR.Pretty.uniform_op
+syntax "remwu" : MLIR.Pretty.uniform_op
+syntax "add.uw" : MLIR.Pretty.uniform_op
+syntax "sh1add.uw" : MLIR.Pretty.uniform_op
+syntax "sh2add.uw" : MLIR.Pretty.uniform_op
+syntax "sh3add.uw" : MLIR.Pretty.uniform_op
+syntax "sh1add" : MLIR.Pretty.uniform_op
+syntax "sh2add" :  MLIR.Pretty.uniform_op
+syntax "sh3add" :  MLIR.Pretty.uniform_op
+
+private def test_simple := [RV64_com| {
+  ^bb0(%e1 : !i64, %e2 : !i64 ):
+  %1 =  add %e1, %e2 : !i64
+        ret %1 : !i64
```

Comment:
```suggestion
  %1 = add %e1, %e2 : !i64
        ret %1 : !i64
```

Why is there a double whitespace? Can you double-check the remainder of the file.

Reply:
@salinhkuhn here, for example, Tobias asks you to check the whole file, so it's nice to reply with "I fixed the whitespace in the whole file!" (assuming you've done so), and hit resolve!